components/profile-controller/pkg/apis/istio/v1alpha1/servicerolebinding_types.go

```diff
@@ -0,0 +1,82 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// ServiceRoleBindingSpec defines the desired state of ServiceRoleBinding
```

Comment:
I don't think we should maintain our istio type.
Istio should provide it, there is an open PR

Reply:
We can use unstructure to manage istio resource in profile controller

Reply:
Which open PR? 
Do they have estimate when to support?

Reply:
We don't need to maintain our istio type, it's one time effort by generating types from Istio CRD.
If we use `unstructured`, for each CRD call we'll need to maintain a struct like https://github.com/kubeflow/kubeflow/blob/master/components/notebook-controller/pkg/controller/notebook/notebook_controller.go#L448
Considering we might need to manage several Istio CRDs, that's non-trivial efforts to make sure all `[]interface{}` match their true CRD types.

Reply:
https://github.com/istio/api/pull/764

Reply:
How did you generate?

Reply:
I generate by
* copy Istio CRD difnition
* code generate via kube builder, like our other CRDs

Seems the PR has been there since Jan. 
We can migrate to it once it's merged and included in istio release.